src/main/java/frc/trigon/robot/constants/SimulatedGamePieceConstants.java

```diff
@@ -0,0 +1,130 @@
+    private static final Rotation3d
+            CORAL_TO_VERTICAL_POSITION_ROTATION = new Rotation3d(0, -Math.PI / 2, 0),
+            CORAL_TO_L2_AND_L3_ROTATION = new Rotation3d(0, Units.degreesToRadians(35), 0);
+
+    public static final ArrayList<SimulatedGamePiece>
+            CORAL_ON_FIELD = new ArrayList<>(List.of(
+            new SimulatedGamePiece(new Pose3d(1.22, FIELD_WIDTH_METERS / 2, 0.15, CORAL_TO_VERTICAL_POSITION_ROTATION), GamePieceType.CORAL),
+            new SimulatedGamePiece(new Pose3d(1.22, FIELD_WIDTH_METERS / 2 - 1.83, 0.15, CORAL_TO_VERTICAL_POSITION_ROTATION), GamePieceType.CORAL),
+            new SimulatedGamePiece(new Pose3d(1.22, FIELD_WIDTH_METERS / 2 + 1.83, 0.15, CORAL_TO_VERTICAL_POSITION_ROTATION), GamePieceType.CORAL),
+            new SimulatedGamePiece(new Pose3d(FIELD_LENGTH_METERS - 1.22, FIELD_WIDTH_METERS / 2, 0.15, CORAL_TO_VERTICAL_POSITION_ROTATION), GamePieceType.CORAL),
+            new SimulatedGamePiece(new Pose3d(FIELD_LENGTH_METERS - 1.22, FIELD_WIDTH_METERS / 2 - 1.83, 0.15, CORAL_TO_VERTICAL_POSITION_ROTATION), GamePieceType.CORAL),
+            new SimulatedGamePiece(new Pose3d(FIELD_LENGTH_METERS - 1.22, FIELD_WIDTH_METERS / 2 + 1.83, 0.15, CORAL_TO_VERTICAL_POSITION_ROTATION), GamePieceType.CORAL)
+    )),
+            ALGAE_ON_FIELD = new ArrayList<>(List.of(
+                    new SimulatedGamePiece(new Pose3d(1.22, FIELD_WIDTH_METERS / 2, 0.5, new Rotation3d()), GamePieceType.ALGAE),
+                    new SimulatedGamePiece(new Pose3d(1.22, FIELD_WIDTH_METERS / 2 - 1.83, 0.5, new Rotation3d()), GamePieceType.ALGAE),
+                    new SimulatedGamePiece(new Pose3d(1.22, FIELD_WIDTH_METERS / 2 + 1.83, 0.5, new Rotation3d()), GamePieceType.ALGAE),
+                    new SimulatedGamePiece(new Pose3d(FIELD_LENGTH_METERS - 1.22, FIELD_WIDTH_METERS / 2, 0.5, new Rotation3d()), GamePieceType.ALGAE),
+                    new SimulatedGamePiece(new Pose3d(FIELD_LENGTH_METERS - 1.22, FIELD_WIDTH_METERS / 2 - 1.83, 0.5, new Rotation3d()), GamePieceType.ALGAE),
+                    new SimulatedGamePiece(new Pose3d(FIELD_LENGTH_METERS - 1.22, FIELD_WIDTH_METERS / 2 + 1.83, 0.5, new Rotation3d()), GamePieceType.ALGAE)
+            ));
+
+    private static final double
+            L2_SCORE_HEIGHT_METERS = 0.70,
+            L3_SCORE_HEIGHT_METERS = 1.1,
+            L4_SCORE_HEIGHT_METERS = 1.75;
+    private static final Rotation3d
+            CORAL_2_OCLOCK_ROTATION = new Rotation3d(0, 0, Math.PI / 3 * 2),
+            CORAL_4_OCLOCK_ROTATION = new Rotation3d(0, 0, Math.PI / 3),
+            CORAL_8_OCLOCK_ROTATION = new Rotation3d(0, 0, -Math.PI / 3),
+            CORAL_10_OCLOCK_ROTATION = new Rotation3d(0, 0, -Math.PI / 3 * 2),
+            CORAL_12_OCLOCK_ROTATION = new Rotation3d(0, 0, Math.PI);
+    private static final Pose3d FIELD_CENTER = new Pose3d(-FIELD_LENGTH_METERS / 2, -FIELD_WIDTH_METERS / 2, 0, new Rotation3d());
+
+    public static final Pose3d[]
```

Comment:
You're not really optimizing anything by having the o'clock rotations, you should ideally have a function to create each array that uses transforms and fors